### PR TITLE
CI matrix change: add Windows Server 2022

### PR DESCRIPTION
**Pull request by aik-jahoda**
OS | What to do | Notes
-- | -- | --
Windows Server 2022 | Add rolling 

part of #57947

This PR changes ssl detection for test for older ssl version

### CODE REVIEW

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs [hunk 1]
```diff
@@ -333,34 +333,59 @@ private static bool GetIsInContainer()
             return (IsLinux && File.Exists("/.dockerenv"));
         }
 
-        private static bool GetSsl3Support()
+        private static bool GetProtocolSupportFormWindowsRegistry(SslProtocols protocol, bool defaultProtocolSupport)
```
Comment (danmoseley) [inline src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:336]:
```suggestion
        private static bool GetProtocolSupportFromWindowsRegistry(SslProtocols protocol, bool defaultProtocolSupport)
```

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs [hunk 2]
```diff
@@ -333,34 +333,59 @@ private static bool GetIsInContainer()
...
+            {
 
                 // Missing key. If we're pre-20H1 then assume SSL3 is enabled.
                 // Otherwise, disabled. (See comments on https://github.com/dotnet/runtime/issues/1166)
                 // Alternatively the returned values must have been some other types.
-                return !IsWindows10Version2004OrGreater;
+                bool ssl3DefaultSupport = !IsWindows10Version2004OrGreater;
+
+#pragma warning disable CS0618 // Ssl2 and Ssl3 are obsolete
+                return GetProtocolSupportFormWindowsRegistry(SslProtocols.Ssl3, ssl3DefaultSupport);
```
Comment (danmoseley) [inline src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:386]:
```suggestion
                return GetProtocolSupportFromWindowsRegistry(SslProtocols.Ssl3, ssl3DefaultSupport);
```

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs [hunk 3]
```diff
@@ -422,25 +453,17 @@ private static bool GetTls13Support()
...
                 // assume no if positive entry is missing on older Windows
                 // Latest insider builds have TLS 1.3 enabled by default.
                 // The build number is approximation.
-                return IsWindows10Version2004Build19573OrGreater;
+                bool defaultProtocolSupport = IsWindows10Version2004Build19573OrGreater;
+
+#if NETFRAMEWORK
+                return false;
+#else
+                return GetProtocolSupportFormWindowsRegistry(SslProtocols.Tls13, defaultProtocolSupport);
```
Comment (danmoseley) [inline src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:464]:
```suggestion
                return GetProtocolSupportFromWindowsRegistry(SslProtocols.Tls13, defaultProtocolSupport);
```

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs [hunk 4]
```diff
@@ -411,7 +441,8 @@ private static bool GetTls11Support()
         private static bool GetTls12Support()
         {
             // TLS 1.1 and 1.2 can work on Windows7 but it is not enabled by default.
-            return !IsWindows7;
+            bool defaultProtocolSupport =  !IsWindows7;
+            return GetProtocolSupportFormWindowsRegistry(SslProtocols.Tls12, defaultProtocolSupport);
```
Comment (danmoseley) [inline src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:445]:
```suggestion
            return GetProtocolSupportFromWindowsRegistry(SslProtocols.Tls12, defaultProtocolSupport);
```

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs [hunk 5]
```diff
@@ -384,21 +409,26 @@ private static bool AndroidGetSslProtocolSupport(SslProtocols protocol)
...
         {
-            // on Windows, macOS, and Android TLS1.0/1.1 are supported.
-            // TLS 1.1 and 1.2 can work on Windows7 but it is not enabled by default.
+            // on Windows, macOS, and Android TLS1.0/1.1 are supported.            
             if (IsWindows)
             {
-                return !IsWindows7;
+                // TLS 1.1 and 1.2 can work on Windows7 but it is not enabled by default.
+                bool defaultProtocolSupport = !IsWindows7;
+                return GetProtocolSupportFormWindowsRegistry(SslProtocols.Tls11, defaultProtocolSupport);
```
Comment (danmoseley) [inline src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:431]:
```suggestion
                return GetProtocolSupportFormWindowsRegistry(SslProtocols.Tls11, defaultProtocolSupport);
```

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs [hunk 6]
```diff
@@ -384,21 +409,26 @@ private static bool AndroidGetSslProtocolSupport(SslProtocols protocol)
...
         {
             // on Windows, macOS, and Android TLS1.0/1.1 are supported.
-            if (IsWindows || IsOSXLike || IsAndroid)
+            if (IsOSXLike || IsAndroid)
             {
                 return true;
+            } 
+            if (IsWindows)
+            {
+                return GetProtocolSupportFormWindowsRegistry(SslProtocols.Tls, true);
```
Comment (danmoseley) [inline src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:418]:
```suggestion
                return GetProtocolSupportFormWindowsRegistry(SslProtocols.Tls, true);
```

**Comment by aik-jahoda** [comment]
@wfurt, this PR change ssl detection for test. Can you review if this change is reasonable?

**Review by wfurt — APPROVED** [review]
Detection changes look ok to me. I like the idea of establishing clear default and generalizing the code.

**Comment by aik-jahoda** [comment]
There is hanging quic test: https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-jajahoda-cimatrix206df4d8ed4ac9495a82/System.Net.Quic.Functional.Tests/1/console.6bf2e8e2.log?sv=2019-07-07&se=2021-12-28T18%3A47%3A50Z&sr=c&sp=rl&sig=UyBQWOi2gRIWUKpm8k300kJpLSOtHw2rT769frEFO%2Bg%3D

cc @ManickaP

**Comment by ericstj** [comment]
Looks like this may have been superseded by https://github.com/dotnet/runtime/pull/64827

@safern @wfurt do we need the PlatofrmDetection changes from this PR?  cc @karelz

**Comment by wfurt** [comment]
I think the detection part is valuable. There is push to disable Tls1.0 and Tls1.1 on Azure machines. While we dodged it for now (hopefully) that day may come so as other can run into it on their machines.

**Comment by safern** [comment]
@wfurt would you mind taking care of that?

**Pull request closed without merging**